js/modules/csv.js

```diff
@@ -0,0 +1,68 @@
+var csvModule = (function(window, $) {
+    function _parseCsvData(data) {
+        return data.split('\n').map(function(row) {
+            var data = row.split('","');
+            return data.map(function(datum) {
+                datum = datum.replace(/"/g, '');
+                datum = datum.replace(/,/g, '. ');
+                return datum;
+            });
+        })
+    }
+
+    function _addcsCategories(rows) {
+        // Asscoate csCategory with csv data on incidntnum
+        var csCategoriesForIncidents = tableModule.csCategoriesForIncidents();
+        return rows.slice(1).map(function(row) {
+            var incidentNum = row[4];
+            var csCategory = csCategoriesForIncidents[incidentNum];
+            row.push(csCategory);
+            return row;
+        });
+    }
+
+    function _formatCsv(rows) {
+        var csvLink = "data:text/csv;charset=utf-8,";
+        return rows.reduce(function(link, row, rowIndex) {
+            link += row.join(",");
+            if (rowIndex < rows.length) {
+                link += "\n";
+            }
+            return link;
+        }, csvLink)
+    }
+
+    function _downloadCsv(dataLink) {
+        var evt = document.createEvent("MouseEvents");
+        evt.initMouseEvent("click", true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);
```

Comment:
Not sure what this is all about. Seems like we're tracking the click to handle the button click. If that's the case, I think there's better was of binding a button click to a js function.

Reply:
Hey @JoeGaebel , sorry for the delayed response.

This lil bit creates a new anchor tag and a click event. It then triggers the click event on the link. For sure there are better ways of binding a button click to a function. The goal of this bit is to name the file, set it's contents, and download it.